core/src/main/java/io/grpc/internal/AbstractServerStream.java

```diff
@@ -149,8 +154,11 @@ protected void inboundHeadersReceived(Metadata headers) {
       }
     }
     if (headers.containsKey(GrpcUtil.MESSAGE_ACCEPT_ENCODING_KEY)) {
-      pickCompressor(
+      Compressor c = pickCompressor(
           ACCEPT_ENCODING_SPLITER.split(headers.get(GrpcUtil.MESSAGE_ACCEPT_ENCODING_KEY)));
+      if (c != null) {
```

Comment:
The semantics of `Stream.pickCompressor()` is unclear when called multiple times. Per current `AbstractStream`, if you call `pickCompressor()` with a list of unsupported encodings, the outcome is different based on whether a compressor has already been set. To avoid the hassle, I would rather forbid calling it multiple times.